docs/admin/kubeadm.md

```diff
@@ -13,188 +13,256 @@ Running `kubeadm init` bootstraps a Kubernetes cluster. This consists of the
+cases you might like to override the default behaviour. Here we specify all the
+flags that can be used to customise the Kubernetes installation.
+
+- `--api-advertise-address`
+
+This is the address the API Server will advertise to other members of the
+cluster.  This is also the address used to construct the suggested `kubeadm
+join` line at the end of the init process.  If not set (or set to 0.0.0.0) then
+IP for the default interface will be used.
+
+This address is also added to the certifcate that the API Server uses.
+
+- `--apiserver-bind-port`
+
+The port that the API server will bind on.  This defaults to 6443.
+
+- `--apiserver-cert-extra-sans`
+
+Additional hostnames or IP addresses that should be added to the Subject
+Alternate Name section for the certificate that the API Server will use.  If you
+expose the API Server through a load balancer and public DNS you could specify
+this with
+
+```
+--apiserver-cert-extra-sans=kubernetes.example.com,kube.example.com,10.100.245.1
+```
+
+- `--cert-dir`
+
+The path where to save and store the certificates.  The default is
+"/etc/kubernetes/pki".
+
+- `--config`
+
+A kubeadm specific [config file](#config-file).  This can be used to specify an
+extended set of options including passing arbitrary command line flags to the
+control plane components.
+
+- `--kubernetes-version` (default 'latest') the kubernetes version to initialise
+
+The **v1.6** version of kubeadm only supports building clusters that are at
+least **v1.6.0**.  There are many reasons for this including kubeadm's use of
+RBAC, the Bootstrap Token system, and enhancements to the Certificates API. With
+this flag you can try any future version of Kubernetes.  Check [releases
+page](https://github.com/kubernetes/kubernetes/releases) for a full list of
+available versions.
 
 - `--pod-network-cidr`
 
 For certain networking solutions the Kubernetes master can also play a role in
-allocating network ranges (CIDRs) to each node. This includes many cloud providers
-and flannel. You can specify a subnet range that will be broken down and handed out
-to each node with the `--pod-network-cidr` flag. This should be a minimum of a /16 so
-controller-manager is able to assign /24 subnets to each node in the cluster.
-If you are using flannel with [this manifest](https://github.com/coreos/flannel/blob/master/Documentation/kube-flannel.yml)
-you should use `--pod-network-cidr=10.244.0.0/16`. Most CNI based networking solutions
-do not require this flag.
+allocating network ranges (CIDRs) to each node. This includes many cloud
+providers and flannel. You can specify a subnet range that will be broken down
+and handed out to each node with the `--pod-network-cidr` flag. This should be a
+minimum of a /16 so controller-manager is able to assign /24 subnets to each
+node in the cluster. If you are using flannel with [this
+manifest](https://github.com/coreos/flannel/blob/master/Documentation/kube-flannel.yml)
```

Comment:
shouldn't we use this the manifest that includes rbac? https://github.com/coreos/flannel/blob/master/Documentation/kube-flannel-rbac.yml